prost-derive/src/lib.rs

```diff
@@ -128,29 +136,36 @@ fn try_message(input: TokenStream) -> Result<TokenStream, Error> {
         }
     });
 
-    let struct_name = if fields.is_empty() {
+    let struct_name = if all_fields.is_empty() {
         quote!()
     } else {
         quote!(
             const STRUCT_NAME: &'static str = stringify!(#ident);
         )
     };
 
-    let clear = fields
+    let clear = all_fields
         .iter()
         .map(|(field_ident, field)| field.clear(quote!(self.#field_ident)));
 
+    // For Default implementation, use all_fields (including skipped)
     let default = if is_struct {
-        let default = fields.iter().map(|(field_ident, field)| {
-            let value = field.default(&prost_path);
+        let default = all_fields.iter().map(|(field_ident, field)| {
+            let value = match field {
+                Field::Skip(skip_field) => skip_field.default_value(),
+                _ => field.default(&prost_path),
```

Comment:
Why are skipped fields handle special in this file? I expect the `Field::Skip` to have the same API as the other fields.

Reply:
Addressed, pushed down to `mod.rs`, similar to `Scalar` field handling.